commit a80c6e71b8a5401fb71b1e79e56ebf565cf785cb (refs/changes/99/3899/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-05-03T18:23:07+02:00 (1 year, 5 months ago)
    
    TOOLS-1224 'sdcadm <subcommand> -h|--help' hits SAPI to get the sdc app: that's overkill

diff --git a/lib/cli/index.js b/lib/cli/index.js
index e3a5b5a..a4efdb2 100644
--- a/lib/cli/index.js
+++ b/lib/cli/index.js
@@ -102,12 +102,14 @@ CLI.prototype.init = function init(opts, args, callback) {
         logComponent = handler.name.slice(3);
     }
     var logToFile = (handler && handler.logToFile || false);
+    var skipInit = false;
 
     if (args.indexOf('help') !== -1 ||
         args.indexOf('--help') !== -1 ||
         args.indexOf('-h') !== -1 ||
         opts.help) {
         logToFile = false;
+        skipInit = true;
     }
 
     // Wrap into try/catch block and handle ENOSPC and EACCES with friendlier
@@ -178,7 +180,7 @@ CLI.prototype.init = function init(opts, args, callback) {
          * manually skip for those here.
          */
         var skipInitCmds = {completion: true, help: true};
-        if (args[0] && skipInitCmds[args[0]]) {
+        if (skipInit || (args[0] && skipInitCmds[args[0]])) {
             callback();
             return;
         }
